pkg/datastore/mysql/ensurer/indexes.sql

```diff
@@ -124,6 +124,10 @@ CREATE INDEX piped_project_id_asc ON Piped (ProjectId);
 -- index on `ProjectId` ASC and `EnvIds` ASC
 ALTER TABLE Piped ADD COLUMN EnvIds JSON GENERATED ALWAYS AS (IFNULL(data ->> "$.env_ids", '[]')) VIRTUAL NOT NULL;
 CREATE INDEX piped_project_id_env_ids_asc ON Piped (ProjectId, (CAST(EnvIds AS CHAR(36) ARRAY)));
+-- Remove the `piped_project_id_env_ids_asc` index due to issue around empty array value on column
+-- which be a part of mulitple columns index.
+-- ref: https://dev.mysql.com/doc/refman/8.0/en/create-index.html#create-index-multi-valued
+DROP INDEX piped_project_id_env_ids_asc ON Piped;
```

Comment:
Does this return an error when the index has already been removed before?
Do we have to check its existence before dropping it?

Reply:
Nice question! In fact, there is no straightforward way to check like `DROP INDEX IF EXISTS` in MySQL 8 😢 There's some workaround available for that, but I chose to keep the `CREATE INDEX` statement as is and add this `DROP INDEX` statement after that, which means
- if this is the first time this ops runs on a newly created datastore, the index will be created first, then remove
- if this is NOT the first time this ops runs on in a used datastore, the `CREATE INDEX` yield `Duplicated` error which will be ignored (handled already), and then this `DROP INDEX` statement is run to remove that existing index
That's the current situation, plz lets me know if you have a better idea for that 🙏

Reply:
Thanks for your deep explanation. I got your point.

How about removing the `CREATE INDEX` and ignoring the error from `DROP INDEX`?

Reply:
That's fine too, then let me update the ensurer to ignore that. TBH, I don't like the current "create then remove" way too, but just think that there're some reasons I don't know that make them abandon the `DROP INDEX IF EXISTS`, and want to return the error on `DROP INDEX` explicitly.

Reply:
Thanks for the idea, addressed by https://github.com/pipe-cd/pipecd/pull/3206/commits/895850ac050fb34f1a8f1ca472e5019853b7f3f8